GUI/src/main/java/ui/UIController.java

```diff
@@ -247,12 +239,18 @@ private void setControlActions() {
         form.tfLink.setOnKeyTyped(e -> processLink());
         form.listView.setOnMouseClicked(e -> {
             if (e.getButton().equals(MouseButton.PRIMARY) && e.getClickCount() == 1) {
-                Job job = form.listView.getSelectionModel().getSelectedItem();
-                if (job != null) {
-                    selectJob(job);
-                    setLink(job.getLink());
-                    setDir(job.getDir());
-                    setFilename(job.getFilename());
+                if (UPDATING_BATCH.getValue().equals(true)) {
+                    clearControls();
+                    setDir(folders.getDownloadFolder());
+                    UPDATING_BATCH.setValue(false);
+                } else {
+                    Job job = form.listView.getSelectionModel().getSelectedItem();
+                    if (job != null) {
+                        selectJob(job);
+                        setLink(job.getSourceLink());
+                        setDir(job.getDir());
+                        setFilename(job.getFilename());
+                    }
```

Comment:
I do not agree with you because refactoring that way would increase lines of code as we are also checking if `UPDATING_BATCH` is set to true or not which is not possible in your proposed solution. Could you please re-analyse the importance of this change?